WordPress/Classes/Models/Notifications/NotificationSettings.swift

```diff
@@ -15,10 +15,20 @@ open class NotificationSettings {
     ///
     public let streams: [Stream]
 
-    /// Maps to the associated blog, if any.
+    /// The associated blog. This property is not thread-safe.
+    ///
+    /// If you need to access this property from a thread different from the one it was created on, don't access it directly. Instead, use the `blogManagedObjectID`
+    /// with `context.existingObject(with: blogManagedObjectID)` to retrieve the corresponding managed object in a thread-safe manner.
     ///
     public let blog: Blog?
 
+    /// The managed object identifier for the associated blog. This property serves as a thread-safe version of the `blog` property.
+    ///
+    /// Rather than directly accessing the `blog` property from a different thread than the one it was created on, use this identifier with
+    /// `context.existingObject(with: blogManagedObjectID)` to retrieve the corresponding managed object in a thread-safe manner.
+    ///
+    public let blogManagedObjectID: NSManagedObjectID?
```

Comment:
More [details](https://github.com/wordpress-mobile/WordPress-iOS/pull/21076/files?diff=unified&w=1#r1263660084) here on why this property was added.

Reply:
Sorry, I'm a bit late to the reviews, but I just noticed it my notifications and wanted to quickly comment on this threading issues.

The comment about the `blog` property applies to any managed object properties. What you would normally want to do if you need to access the `blog` property from a different thread is re-fetch the `notifications` object itself first:

```swift
let settingsID = settings.objectID
context.perform {
    let settings = try? context.existingObject(with: settingsID)
    let blog = settings.blog // Now you can access the associated blog safely
}
```

The main advantage is that this way there is no need to duplicate data (`blogManagedObjectID`).

Reply:
You're right, Kean. If `NotificationSettings` was a managed object, any of its properties, like `blogManagedObjectID`, wouldn't be safe to access from a different thread. But in this situation, `NotificationSettings` isn't a managed object, so it's safe to access its properties except the `blog` property.

Reply:
Oh, you are right. I knew I misunderstood something. I just assumed that was also a managed object. 

Is there a way to remove the `blog` property entirely and keep only the new `blogManagedObjectID` property?

Reply:
> Is there a way to remove the blog property entirely and keep only the new blogManagedObjectID property?

Yes, that's the plan. See this [discussion](https://github.com/wordpress-mobile/WordPress-iOS/pull/21076#discussion_r1266013556).

Reply:
Good stuff!